core/src/com/unciv/ui/cityscreen/CityScreen.kt

```diff
@@ -134,20 +135,24 @@ class CityScreen(internal val city: CityInfo): CameraStageBaseScreen() {
                 city.annexCity()
                 update()
             }
+            if (viewingCiv.isSpectator()) annexCityButton.disable()
             razeCityButtonHolder.add(annexCityButton).colspan(cityPickerTable.columns)
         } else if(!city.isBeingRazed) {
             val razeCityButton = "Raze city".toTextButton()
             razeCityButton.labelCell.pad(10f)
             razeCityButton.onClick { city.isBeingRazed=true; update() }
-            if(!UncivGame.Current.worldScreen.isPlayersTurn || city.isOriginalCapital)
+            if(!UncivGame.Current.worldScreen.isPlayersTurn
```

Comment:
There seem to be a lot of 'is players turn and is not spectator' checks, that should probably be put in its own val
Maybe call is canChangeState or something

Reply:
Great idea! Added it to worldScreen and it simplified a lot of those "can I add/enable state changing button" checks.